### PR TITLE
Analytics & Tracking for sign up flows on public library pages

### PR DESCRIPTION
**Pull request by ahsu1230**
@andrewconner @yipingliao

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/libraries/library.js [hunk 1]
```diff
@@ -61,7 +64,11 @@ angular.module('kifi')
         $rootScope.$emit('trackLibraryEvent', 'view', { type: 'installLibrary' });
       }
 
-      $scope.close = modalService.close;
+      $scope.close = function () {
+        $analytics.eventTrack('visitor_clicked_page', {type : 'installLibrary', action: 'close'});
+        return modalService.close;
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/libraries/library.js:69]: Same comment here as what we just discussed. Hmm...

File: kifi-backend/modules/shoebox/angular/src/signup/signupService.js [hunk 2]
```diff
@@ -260,7 +288,11 @@ angular.module('kifi')
           } else if (installService.isValidFirefox) {
             $scope.platformName = 'Firefox';
           }
-          $scope.installExtension = installService.triggerInstall;
+          $scope.installExtension = function() {
+            $analytics.eventTrack('visitor_clicked_page', {type: 'installLibrary', action: 'install'});
+            $scope.installTriggered = true;
+            return installService.triggerInstall;
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/signup/signupService.js:294]: And here. I am curious as to how it works. Maybe it was not working before?!

File: kifi-backend/modules/shoebox/angular/src/signup/signupService.js [hunk 3]
```diff
@@ -126,11 +138,17 @@ angular.module('kifi')
...
+        $rootScope.$emit('trackLibraryEvent', 'click', { type: 'signupLibrary', action: 'clickAuthTwitter' });
+      }
+    };
+
     $scope.fbAuthFromLibrary = function () {
       if ($scope.userData.libraryId) {
         $rootScope.$emit('trackLibraryEvent', 'click', { type: 'signupLibrary', action: 'clickAuthFacebook' });
       }
-      return $scope.fbAuth();
+      //return $scope.fbAuth(); //unused
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/signup/signupService.js:151]: @ahsu1230  Is the return value not used, or is the method call redundant? In either case, let’s remove the commented out code.

Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/signup/signupService.js:151]: Ok, I was hesitant in removing all the FB auth scripts, but I guess if it'll be saved on Github anyway, we could always return back to the FB Auth popup. @andrewconner

**Pull request merged**